src/haxelib/api/LibFlagData.hx

```diff
@@ -63,9 +63,13 @@ private final libraryFlagEReg = ~/^-(lib|L|-library)\b/;
 	Extracts the lib information from the hxml file at `path`.
 
 	Does not filter out repeated libs.
+
+	Target libraries (e.g. hxcpp) are also added if they are not explicitly
+	included via a -lib flag.
  **/
-function fromHxml(path:String):List<{name:ProjectName, data:Option<VersionData>, isTargetLib:Bool}> {
-	final libsData = new List<{name:ProjectName, data:Option<VersionData>, isTargetLib:Bool}>();
+function fromHxml(path:String):List<{name:ProjectName, data:Option<VersionData>}> {
+	final libsData = new List<{name:ProjectName, data:Option<VersionData>}>();
+	final targetLibraries = [for (_ => lib in TARGETS) lib => null];
```

Comment:
Does this infer correctly on Haxe 4.2? I'm not sure if this ends up being `Null<Bool>` there, which would make this tri-state not work. Makes no difference on neko, but could be relevant if we ever use haxelib on a static target.

Reply:
Good catch.